pkg/reconcilermanager/controllers/garbage_collector.go

```diff
@@ -131,17 +129,16 @@ func (r *RepoSyncReconciler) deleteServiceAccount(ctx context.Context, reconcile
 	return r.cleanup(ctx, reconcilerRef, kinds.ServiceAccount())
 }
 
-func (r *RepoSyncReconciler) deleteRoleBinding(ctx context.Context, rsList *v1beta1.RepoSyncList, rsNamespace, reconcilerNamespace string) error {
-	rbName := RepoSyncPermissionsName()
-	if len(rsList.Items) == 0 {
-		return r.cleanup(ctx, client.ObjectKey{Namespace: rsNamespace, Name: rbName}, kinds.RoleBinding())
-	}
+func (r *RepoSyncReconciler) deleteRoleBinding(ctx context.Context, reconcilerRef, rsRef types.NamespacedName) error {
+	rbKey := client.ObjectKey{Namespace: rsRef.Namespace, Name: RepoSyncPermissionsName()}
 	rb := &rbacv1.RoleBinding{}
-	if err := r.client.Get(ctx, client.ObjectKey{Namespace: rsNamespace, Name: rbName}, rb); err != nil {
-		return errors.Wrapf(err, "failed to get the RoleBinding object %s/%s", rsNamespace, rbName)
+	if err := r.client.Get(ctx, rbKey, rb); err != nil {
+		return errors.Wrapf(err, "failed to get the RoleBinding object %s", rbKey)
 	}
-	if err := r.updateRoleBindingSubjects(rb, rsList, reconcilerNamespace); err != nil {
-		return errors.Wrapf(err, "failed to update subjects of the RoleBinding object %s/%s", rsNamespace, rbName)
+	rb.Subjects = removeSubject(rb.Subjects, r.serviceAccountSubject(reconcilerRef))
+	if len(rb.Subjects) == 0 {
+		// Delete the whole RB
+		return r.cleanup(ctx, rbKey, kinds.RoleBinding())
```

Comment:
Is it necessary to delete the RoleBinding?

Reply:
This is the existing behavior when all the subjects are removed. Previously it was just implemented after the LIST of RSyncs, instead of after the GET of the RoleBinding.